modules/api/src/main/java/org/apache/ignite/tx/IgniteTransactions.java

```diff
@@ -84,6 +84,8 @@ public interface IgniteTransactions {
     /**
      * Executes a closure within a transaction.
      *
+     * <p>Please make sure all asynchronous operations are enlisted into the transaction before returning from the callback.
```

Comment:
I'm sure that user won't understand this. I don't even believe that user knows what enlist means.
I'd rather enforce user with javadoc to use only sync operations or anync ones with joins or gets.

Reply:
**I'm sure that user won't understand this** - why do you think a user is stupid ? This sentence is rather clear. I can rephrase it, but the current approach seems OK. You can provide your own variant to evaluate.

**I'd rather enforce user with javadoc to use only sync operations or anync ones with joins or gets** - this is not that we want. The API allows mixing sync and async ops in a free way. It is the intentional design choice and was promised to a user. Let's not break this statement.

Reply:
> why do you think a user is stupid ? 
I don't think that the user is stupid. However proposed javadoc isn't explicit enough.
- What does enlist means?
- How can I, as a user, be sure that all operations are enlisted?
- How should I, as a user, properly write opAsync.handle clauses? Is it possible?
- etc.
> this is not that we want. The API allows mixing sync and async ops in a free way. It is the intentional design choice and was promised to a user. Let's not break this statement.
Let me rephrase 
I'd rather enforce user with javadoc to use **either** sync operations **or** anync ones with joins or gets **or the combinations of aforementioned** Let's also advice user to use runInTransactionAsync for asyncOps chaining, providing several examples or proper thenCompose, etc and illegal thenApply, etc usecases.
Generally speaking, it's a public non trivial async api, so that, being as explicit with javadoc as possible, deffinitely worth a short.